### ISSUE TITLE
[Question] Custom Packet Creation

### ISSUE THREAD

**Issue by wreighsantos** [basil00/Divert#139]
Hi, I have a question.

Is it possible to create my own packet and send it via WinDivertSend? If so, how can I do it?
I suppose I can create my own byte array for the packet, but what about the Header structs? For what I've experienced in using WinDivert, the header structs are initialized inside ParsePacket. 
So can I just pass, for instance, an empty packet to ParsePacket?
Because I need to create an initial packet that can be modified later on and then be sent.
Example flow:
- Init packet -> create empty packet array and parse the packet
- Update data -> update the data of the parsed packet
- Update some headers -> update respective header structs from parsed packet
- Send -> send parsed packet data (which was updated in step 2)

Will this work?
Thank you!

**Comment by basil00** [comment]
Yes you can craft and inject a packet from scratch.  It is just a matter of creating a byte array that is a valid packet.  One way is to create a `struct` with headers as members, as the `blockpage` packet from the `webfilter` sample.  One must be careful that the compiler does not introduce padding between the headers though.  The `ParsePacket` function only parses existing packets, and does not help in the construction of new packets.

**Comment by wreighsantos** [comment]
Ok, sorry for the inconvenience. Next time I will post in stackoverflow. Thanks a lot!
